### ISSUE TITLE
Add a generator for NSX-T

### ISSUE THREAD

**Issue by jcatrouillet** [google/capirca#330]
There's a generator for NSX-v, but this product is officially not supported anymore and NSX-T is its successor.
I'm working on creating this generator, this is my very first contribution to an open-source project, so any guidance would be helpful.

There's a preliminary version on my private repo:
https://github.com/jcatrouillet/capirca

I still have few things to handle before submitting a PR.

Any feedback would be welcome!

**Comment by ivucica** [comment]
There is work being done in https://github.com/google/capirca/pull/338.

**Comment by ivucica** [comment]
Your contribution is interesting as well, and it might be interesting to compare the outputs of the two generators.

Of course, if you open a PR when #338 lands, your PR will likely conflict, but it might be possible to work around that -- merging the two contributions. 

I'm interested in #338 landing because there's more work being done on a pusher based on the NSX-T API, and #338 is meant to work with that; however, it would certainly be interesting to compare the output of the generators and see if they can be merged into one.

Would you consider rebasing your code once #338 lands, signing the CLA [here](https://cla.developers.google.com/about/google-individual) (or, if contributing as a corporation, [here](https://cla.developers.google.com/about/google-corporate)) and opening a PR?

**Comment by jcatrouillet** [comment]
Hi Ivan, I quickly looked at #338 and I see few shortcomings already, like the length of the comments not managed, the protocols are limited to ICMP, ICMPv6, TCP and UDP, while NSX-T accepts raw numbers. I cannot find how exclude address is managed. 
I did not create a PR as I didn't get any feedback to this issue and I found my tests too light, but looking at #338 I guess I was too perfectionist :)
